### PR TITLE
 Plumb message validator to sender and validate outgoing messages before adding them to mpool.

### PR DESCRIPTION
**Pull request by anorth**
This is most of the work for #1934 (there are some cases the validator doesn't yet check).

I intend to rebase on #2136 and #2138 before merging, resolving some duplicated actor loading.

### CODE REVIEW

File: plumbing/msg/sender.go [hunk 1]
```diff
@@ -56,22 +55,38 @@ func (s *Sender) Send(ctx context.Context, from, to address.Address, value *type
 	s.l.Lock()
 	defer s.l.Unlock()
 
-	nonce, err := nextNonce(ctx, s.chainReader, s.msgPool, from)
+	st, err := s.chainReader.LatestState(ctx)
+	if err != nil {
+		return cid.Undef, err
```
Comment (phritz) [inline plumbing/msg/sender.go]: could be wrapped for more clarity, eg "couldnt get latest state: " here and below

File: plumbing/msg/sender.go [hunk 2]
```diff
@@ -81,19 +96,13 @@ func (s *Sender) Send(ctx context.Context, from, to address.Address, value *type
...
 // nextNonce returns the next nonce for the given address. It checks
 // the actor's memory and also scans the message pool for any pending
 // messages.
-func nextNonce(ctx context.Context, chainReader chain.ReadStore, msgPool *core.MessagePool, address address.Address) (nonce uint64, err error) {
-	st, err := chainReader.LatestState(ctx)
-	if err != nil {
-		return 0, err
-	}
-
+func nextNonce(ctx context.Context, st state.Tree, msgPool *core.MessagePool, address address.Address) (nonce uint64, err error) {
```
Comment (frrist) [inline plumbing/msg/sender.go]: nit: not a fan of named returns for a method like this.

**Comment by anorth** [comment]
It seems I was premature putting this up to review. The test failure is non-trivial. I'll ping again for review if I fold the improved test support in here (but will try to do it in a different PR).

**Comment by anorth** [comment]
@phritz please take another quick squiz before I land. The major new change is test setup in `message_propagate_test.go`. Note that there's also a flag on the validator now, opting out of the nonce-too-high check for outbound messages.

**Pull request merged**